unit-tests/src/test/scala/scala/scalanative/unsafe/PtrBoxingTest.scala

```diff
@@ -198,13 +198,14 @@ class PtrBoxingTest {
 
       val testStr        = toCString("hello_native")
       val expectedInt    = 42
-      val expectedLength = 12
+      val expectedLength = 12.toULong
 
       assertEquals(expectedInt, x._1.apply())
       assertEquals(expectedInt, loadedGetInt())
 
-      assertEquals(expectedLength, x._2.apply(testStr))
-      assertEquals(expectedLength, loadedStringLength(testStr))
+      //assertEquals uses Object.equals which would return false in following cases
```

Comment:
That seems very problematic. Two equivalent `ULong`s should compare equal according to `Object.equals`. It seems we should override `equals` and `hashCode` in all the `U` types. It should be in a separate PR, so please file an issue and reference from here (as well as from the other places where `assertEquals` was replaced by `assert`).

Reply:
I guess this PR can now be rebased and amended to go back to using `assertEquals`, now that #2075 was merged.